superset/viz.py

```diff
@@ -694,15 +694,14 @@ def get_data(self, df: pd.DataFrame) -> VizData:
         columns = self.form_data.get("columns")
         if self.form_data.get("transpose_pivot"):
             groupby, columns = columns, groupby
+        metrics = [utils.get_metric_name(m) for m in self.form_data["metrics"]]
         df = df.pivot_table(
             index=groupby,
             columns=columns,
-            values=[
-                utils.get_metric_name(m) for m in self.form_data.get("metrics", [])
```

Comment:
There's no need to do `self.form_data.get("metrics", [])` since in the `query_obj` method we've already verified there exists at least one metric.